gestalt-module/src/main/java/org/terasology/module/ModuleMetadataJsonAdapter.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.module;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.reflect.TypeToken;
+
+import org.terasology.i18n.I18nMap;
+import org.terasology.i18n.gson.I18nMapTypeAdapter;
+import org.terasology.naming.Name;
+import org.terasology.naming.Version;
+import org.terasology.naming.gson.NameTypeAdapter;
+import org.terasology.naming.gson.VersionTypeAdapter;
+
+import java.io.Reader;
+import java.io.Writer;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Reads/writes ModuleMetadata to and from a json format.
+ * An example of this format is:
+ * <pre>
+ *     {
+ *          "id": "Core",
+ *          "version": "0.1.0",
+ *          "displayName": {
+ *              "en": "Short Name"
+ *          },
+ *          "description": {
+ *              "en": "A longer description of the module"
+ *          },
+ *          "dependencies": [
+ *              {
+ *                  "id": "baseModule",
+ *                  "minVersion": "1.0.0",
+ *                  "maxVersion": "2.0.0"
+ *              }
+ *          ]
+ *     }
+ * </pre>
+ * Extensions can be registered, as a mapping of identifier to Type, and optionally with a JsonSerializer/JsonDeserializer or TypeAdapter.
+ *
+ * @author Immortius
+ */
+public class ModuleMetadataJsonAdapter {
+
+    private static final Type DEPENDENCY_LIST_TYPE = new TypeToken<List<DependencyInfo>>() {
+    }.getType();
+    private static final Type STRING_SET_TYPE = new TypeToken<Set<String>>() {
+    }.getType();
+
+    private final GsonBuilder builder;
+    private transient Gson cachedGson;
+    private final Map<String, Type> extensionMap = Maps.newHashMap();
+
+    public ModuleMetadataJsonAdapter() {
+        this.builder = new GsonBuilder()
+                .setPrettyPrinting()
+                .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
```

Comment:
Both are equally valid. The `Z` means zero offset  from UTC time. The `XXX` uses the (long format) timezone offset (e.g. -05:00). [Wikipedia has some details on the specs.](
http://en.wikipedia.org/wiki/ISO_8601#UTC). 
The `Z` variant silently assumes that `Date` is in UTC time, which is a potential source of error, imho.